### PR TITLE
Don't exit immediately on error in lint script

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Prevent `lint.sh` from failing fast.

## How is this patch tested?

Injected a dummy lint error and verified the new script works as I expected.

https://github.com/mlflow/mlflow/pull/3349/checks?check_run_id=1039704095#step:4:1

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: lint.sh [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #!/usr/bin/env bash
 
-set -e
+# Set err=1 if any commands exit with non-zero status as described in
+# https://stackoverflow.com/a/42219754
+err=0
+trap 'err=1' ERR
```
Comment (harupy) [inline lint.sh:6]: Used the same approach as https://github.com/mlflow/mlflow/blob/master/dev/run-python-flavor-tests.sh

File: mlflow/sklearn/__init__.py [hunk 2]
```diff
@@ -792,7 +792,7 @@ def _log_posttraining_metadata(estimator, *args, **kwargs):
             try:
                 score_args = _get_args_for_score(estimator.score, estimator.fit, args, kwargs)
                 training_score = estimator.score(*score_args)
-            except Exception as e:  # pylint: disable=broad-except
+            except Exception as e:
```
Comment (harupy) [inline mlflow/sklearn/__init__.py:795]: I'll revert this change after veryfing the new script works as expected.

**Comment by harupy** [comment]
The error log looks like this:

Screen Shot 2020-10-21 at 11 44 08

**Review by dbczumar — APPROVED** [review]
LGTM!

**Comment by andrewnitu** [comment]
maybe if `test err=0` fails we can have a message saying "one of the previous steps failed, check above" just so it is clear for OSS authors

**Comment by harupy** [comment]
@andrewnitu 

> maybe if test err=0 fails we can have a message saying "one of the previous steps failed, check above" just so it is clear for OSS authors

Sounds good. I'll add the message at the end of the script.

**Pull request merged**